components/camel-kafka/src/main/java/org/apache/camel/component/kafka/CamelKafkaExchangeEncoder.java

```
@@ -0,0 +1,53 @@
+package org.apache.camel.component.kafka;
+
+import kafka.serializer.Encoder;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+
+public class CamelKafkaExchangeEncoder<E extends Encoder<V>, V> implements Encoder<CamelKafkaGenericObject<V>> {
+
+  E clientEncoder;
+
+  public void setClientEncoder(E clientEncoder) {
+    this.clientEncoder = clientEncoder;
+  }
+
+  @Override
+  public byte[] toBytes(CamelKafkaGenericObject<V> vCamelKafkaGenericObject) {
+    try {
+      byte[] bytes = clientEncoder.toBytes(vCamelKafkaGenericObject.getBody());
+      CamelKafkaGenericObject camelKafkaGenericObject =
+          new CamelKafkaGenericObject.CamelKafkaGenericObjectBuilder<byte[]>()
+              .setBody(bytes)
+              .setHeaders(vCamelKafkaGenericObject.getHeaders())
+              .build();
+      return getBytes(camelKafkaGenericObject);
+    } catch (IOException e) {
+      e.printStackTrace();
+      return null;
```

Lets do log.error.

Again, shall we throw exception instead of sending a null to kafka?